pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/ViewFilterRoute.java

```diff
@@ -67,7 +69,14 @@ public Object processRequest(Request request, Response response, String userId)
                 }
             } else if (request.url().contains(RoutePath.SAVE_FILTER)) {
                 String ddpGroupId = DDPInstance.getDDPGroupId(realm);
-                return ViewFilter.saveFilter(json, userIdRequest, patchUtil.getColumnNameMap(), ddpGroupId);
+                ViewFilter viewFilter = new Gson().fromJson(json, ViewFilter.class);
+
+                String filterName = viewFilter.getFilterName();
```

Comment:
It might make sense to trim `filterName` and check for empty.
Not sure of the request body origin, but we've definitely seen problems with extra whitespace in input strings.

Reply:
Agreed, will apply this inside of `FilterView` state so that it's trimmed when constructed.